internal/cmd/beta/network-interface/update/update.go

```diff
@@ -0,0 +1,242 @@
+)
+
+type inputModel struct {
+	*globalflags.GlobalFlagModel
+	NicId            string
+	NetworkId        *string
+	AllowedAddresses *[]iaas.AllowedAddressesInner
+	Labels           *map[string]string
+	Name             *string // <= 63 characters + regex  ^[A-Za-z0-9]+((-|_|\s|\.)[A-Za-z0-9]+)*$
+	NicSecurity      *bool
+	SecurityGroups   *[]string // = 36 characters + regex ^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$
+}
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: "Updates a network interface",
+		Long:  "Updates a network interface.",
+		Args:  args.SingleArg(nicIdArg, utils.ValidateUUID),
+		Example: examples.Build(
+			examples.NewExample(
+				`Updates a network interface with nic id "xxx" and network-id "yyy" with new allowed addresses "1.1.1.1,8.8.8.8,9.9.9.9" and with new label "key=value"`,
+				`$ stackit beta network-interface update xxx --network-id yyy --allowed-addresses "1.1.1.1,8.8.8.8,9.9.9.9" --labels key=value"`,
+			),
+			examples.NewExample(
+				`Updates a network interface with nic id "xxx" and network-id "yyy" with new name "nic-name-new"`,
+				`$ stackit beta network-interface update xxx --network-id yyy --name nic-name-new"`,
+			),
+			examples.NewExample(
+				`Updates a network interface with nic id "xxx" and network-id "yyy" with security groups removed and with nic-security disabled`,
+				`$ stackit beta network-interface update xxx --network-id yyy --nic-security=false --security-groups """`,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			model, err := parseInput(p, cmd, args)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to update the network interface %q?", model.NicId)
+				err = p.PromptForConfirmation(prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("create network area: %w", err)
+			}
+
+			return outputResult(p, model, resp)
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Var(flags.UUIDFlag(), networkIdFlag, "Network ID")
+	cmd.Flags().StringSlice(allowedAddressesFlag, nil, "List of allowed IPs")
+	cmd.Flags().StringToString(labelFlag, nil, "Labels are key-value string pairs which can be attached to a network-interface. A label can be provided with the format key=value and the flag can be used multiple times to provide a list of labels")
+	cmd.Flags().StringP(nameFlag, "n", "", "Network interface name")
+	cmd.Flags().BoolP(nicSecurityFlag, "b", defaultEnabled, "If this is set to false, then no security groups will apply to this network interface.")
```

Comment:
I think that `defaultNicSecurityFlag` is a better name than `defaultEnabled` here